### PR TITLE
Switch more of the linalg tests to use testsets

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by kshyatt** [comment]
cc @andreasnoack have I kept the spirit of your tests intact?

File: test/linalg/arnoldi.jl [hunk 1]
```diff
@@ -150,87 +152,91 @@ let
...
+            s1_left = sign(S1[1][:U][3,1]) * S1[1][:U][:,1]
+            s2_left = sign(S2[1][3,1]) * S2[1][:,1]
+            @test s1_left ≈ s2_left
+
+            ## 1st right singular vector
+            s1_right = sign(S1[1][:Vt][3,1]) * S1[1][:Vt][:,1]
+            s2_right = sign(S2[3][3,1]) * S2[3][:,1]
+            @test s1_right ≈ s2_right
+        end
+        @testset "10329" begin
```
Comment (tkelman) [inline test/linalg/arnoldi.jl]: just a number with no context isn't very useful

Comment (kshyatt) [inline test/linalg/arnoldi.jl]: Yeah true I'll put more info there

File: test/linalg/diagonal.jl [hunk 2]
```diff
@@ -180,26 +186,30 @@ for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})
...
 
-    #10036
-    @test issymmetric(D2)
-    @test ishermitian(D2)
-    if elty <: Complex
-        dc = d + im*convert(Vector{elty}, ones(n))
-        D3 = Diagonal(dc)
-        @test issymmetric(D3)
-        @test !ishermitian(D3)
+    @testset "10036" begin
```
Comment (tkelman) [inline test/linalg/diagonal.jl]: likewise here

**Comment by kshyatt** [comment]
Added more info and CI has passed.

**Pull request merged**